### PR TITLE
Update icinga2-commands.conf

### PR DESCRIPTION
**Pull request by pdorschner**
Updates the icinga2-commands.conf with new subcommands and examples

### CODE REVIEW

File: contrib/icinga2-commands.conf [hunk 1]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+object CheckCommand "cloud_aws_s3_bucket" {
+    import "cloud_aws"
+
+    command += [ "s3", "bucket" ]
 
-    vars.cloud_aws_ec2 = "ec2"
-    vars.loud_aws_instances = "instances"
+    arguments += {
+        "-b" = {
+            value = "$cloud_aws_s3_bucket_buckets$"
```
Comment (dgoetz) [inline contrib/icinga2-commands.conf:78]: `cloud_aws_s3_bucket_buckets` is not a nice name, so while I understand the schema, repeating bucket seams unnecessary. Personal preference would be `cloud_aws_s3_buckets`.

File: contrib/icinga2-commands.conf [hunk 2]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+            repeat_key = true
+            description = "Name of the S3 bucket. If --buckets is empty, all buckets will be evaluated"
+        }
+        "-c" = {
+            value = "$cloud_aws_s3_bucket_crit_bucket_size$"
```
Comment (dgoetz) [inline contrib/icinga2-commands.conf:83]: `cloud_aws_s3_bucket_crit_bucket_size` is not a nice name, so while I understand the schema, repeating bucket seams unnecessary. Personal preference would be `cloud_aws_s3_bucket_crit_size`.

File: contrib/icinga2-commands.conf [hunk 3]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+            description = "Critical threshold for the size of the specified bucket. Alerts if the size is greater than the critical threshold"
+        }
+        "-w" = {
+             value = "$cloud_aws_s3_bucket_warn_bucket_size$"
```
Comment (dgoetz) [inline contrib/icinga2-commands.conf:87]: `cloud_aws_s3_bucket_warn_bucket_size` is not a nice name, so while I understand the schema, repeating bucket seams unnecessary. Personal preference would be `cloud_aws_s3_bucket_warn_size`.

File: contrib/icinga2-commands.conf [hunk 4]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+             description = "Warning threshold for the size of the specified bucket. Alerts if the size is greater than the warning threshold"
+        }
+        "--empty-ok" = {
+            set_if = "$cloud_aws_s3_bucket_empty_ok$"
+            description = "Return OK if there are no buckets"
```
Comment (dgoetz) [inline contrib/icinga2-commands.conf:92]: Empty ok and no buckets sound like to different states. Can this be made more clear?

File: contrib/icinga2-commands.conf [hunk 5]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+object CheckCommand "cloud_aws_s3_object" {
+    import "cloud_aws"
+
+    command += [ "s3", "object" ]
+
+    arguments += {
+        "-b" = {
+            value = "$cloud_aws_s3_object_bucket$"
+            repeat_key = true
+            description = "Name of the S3 bucket. If --buckets is empty, all buckets will be evaluated"
```
Comment (dgoetz) [inline contrib/icinga2-commands.conf:106]: `--buckets` makes no sense in the description as the user should not see this, especially when -b is used as argument. Better use the value `cloud_aws_s3_object_bucket` or something like "If no bucket is given, ...".

**Review by martialblog — COMMENTED** [review]
LGTM :+1:

File: contrib/icinga2-commands.conf [hunk 6]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+        }
+        "--empty-ok" = {
+            set_if = "$cloud_aws_s3_bucket_empty_ok$"
+            description = "Return OK if there are no buckets"
+        }
+    }
 }
 
-apply Service "cloud_aws_single_instance" {
+object CheckCommand "cloud_aws_s3_object" {
```
Comment (dgoetz) [inline contrib/icinga2-commands.conf:97]: I do not understand the difference between this and `cloud_aws_s3_bucket`, as it only looks like the other handles more buckets at once?

File: contrib/icinga2-commands.conf [hunk 7]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+apply Service "cloud_aws_status_example" {
+    import "generic-service"
+
+    check_command = "cloud_aws_status"
+
+    vars.cloud_aws_status_url = "https://status.aws.amazon.com"
+    vars.cloud_aws_status_service = "cloudwatch"
+
+    assign where host.name == NodeName
+}
```
Comment (dgoetz) [inline contrib/icinga2-commands.conf:205]: There should be a new line at the end of file.

File: contrib/icinga2-commands.conf [hunk 8]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+    arguments += {
+        "-u" = {
+            value = "$cloud_aws_status_url$"
+            description = "The AWS Status Page URL"
+        }
+        "-s" = {
+            value = "$cloud_aws_status_service$"
+            description = "The AWS Service to check"
+        }
+    }
```
Comment (dgoetz) [inline contrib/icinga2-commands.conf:145]: The example below looks like there is a valid default for the status URL, can this be added here?

File: contrib/icinga2-commands.conf [hunk 9]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+            description = "Limits the response to keys that begin with the specified prefix"
+        }
+        "-p" = {
+            set_if = "$cloud_aws_s3_object_perfdata$"
+            description = "Displays perfdata and lists all objects in the specified bucket"
+        }
+        "--empty-ok" = {
+            set_if = "$cloud_aws_s3_object_empty_ok$"
+            description = "Return OK if there are no buckets"
+        }
```
Comment (dgoetz) [inline contrib/icinga2-commands.conf:127]: Would defaults for perfdata handling and empty make sense?

File: contrib/icinga2-commands.conf [hunk 10]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+        "-w" = {
+             value = "$cloud_aws_s3_bucket_warn_bucket_size$"
+             description = "Warning threshold for the size of the specified bucket. Alerts if the size is greater than the warning threshold"
+        }
+        "--empty-ok" = {
+            set_if = "$cloud_aws_s3_bucket_empty_ok$"
+            description = "Return OK if there are no buckets"
+        }
+    }
```
Comment (dgoetz) [inline contrib/icinga2-commands.conf:94]: Would defaults for empty make sense?

**Review by dgoetz — CHANGES_REQUESTED** [review]

File: contrib/icinga2-commands.conf [hunk 11]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
-            description = "Checks a single EC2 instance"
-        }
         "-i" = {
-            value = "$cloud_aws_id$"
+            value = "$cloud_aws_ec2_instance_id$"
             description = "Look for ec2 instance by id"
         }
         "-n" = {
-            value = "$cloud_aws_name$"
+            value = "$cloud_aws_ec2_instance_name$"
```
[the remark this answers is not in the document]
Comment (pdorschner) [reply contrib/icinga2-commands.conf:35]: This is specific to the check command instance

File: contrib/icinga2-commands.conf [hunk 12]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
-            value = "$cloud_aws_ec2$"
-            description = "Checks in the EC2 context"
-        }
-        "instances" = {
-            value = "$cloud_aws_instances$"
-            description = "Checks multiple EC2 instances by name"
-        }
         "-n" = {
-            value = "$cloud_aws_name$"
+            value = "$cloud_aws_ec2_instances_name$"
```
[the remark this answers is not in the document]
Comment (pdorschner) [reply contrib/icinga2-commands.conf:48]: This is specific to the check command instances

File: contrib/icinga2-commands.conf [hunk 13]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
             description = "Search for ec2 instances by name (e.g. instance*)"
         }
         "-a" = {
-            value = "$cloud_aws_autoscale$"
+            value = "$cloud_aws_ec2_instances_autoscale$"
```
[the remark this answers is not in the document]
Comment (pdorschner) [reply contrib/icinga2-commands.conf:52]: This is  specific to the check command instances

File: contrib/icinga2-commands.conf [hunk 14]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+
+object CheckCommand "cloud_aws_cloudfront" {
+    import "cloud_aws"
+
+    command += [ "cloudfront" ]
+
+    arguments += {
+        "-e" = {
+            value = "$cloud_aws_etag$"
+            description = "Etag name of one or multiple distributions. If --etag is empty, all distributions will be evaluated"
```
[the remark this answers is not in the document]
Comment (pdorschner) [reply contrib/icinga2-commands.conf]: Done

File: contrib/icinga2-commands.conf [hunk 15]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
+
+    command += [ "s3", "bucket" ]
 
-    vars.cloud_aws_ec2 = "ec2"
-    vars.loud_aws_instances = "instances"
+    arguments += {
+        "-b" = {
+            value = "$cloud_aws_s3_bucket_buckets$"
+            repeat_key = true
+            description = "Name of the S3 bucket. If --buckets is empty, all buckets will be evaluated"
```
[the remark this answers is not in the document]
Comment (pdorschner) [reply contrib/icinga2-commands.conf]: Done

File: contrib/icinga2-commands.conf (hunk 1 above)
Comment (pdorschner) [reply contrib/icinga2-commands.conf]: Done

File: contrib/icinga2-commands.conf (hunk 2 above)
Comment (pdorschner) [reply contrib/icinga2-commands.conf]: Done

File: contrib/icinga2-commands.conf (hunk 3 above)
Comment (pdorschner) [reply contrib/icinga2-commands.conf]: Done

File: contrib/icinga2-commands.conf (hunk 6 above)
Comment (pdorschner) [reply contrib/icinga2-commands.conf:97]: This is specific for the objects inside a bucket

File: contrib/icinga2-commands.conf (hunk 7 above)
Comment (pdorschner) [reply contrib/icinga2-commands.conf]: Done

File: contrib/icinga2-commands.conf (hunk 8 above)
Comment (pdorschner) [reply contrib/icinga2-commands.conf:145]: Done

File: contrib/icinga2-commands.conf (hunk 10 above)
Comment (pdorschner) [reply contrib/icinga2-commands.conf:94]: Per default, if no buckets were found, it returns UNKNOWN. The use can decide if it returns OK or UNKNOWN

File: contrib/icinga2-commands.conf [hunk 16]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
-            value = "$cloud_aws_ec2$"
-            description = "Checks in the EC2 context"
-        }
-        "instance" {
-            value = "$cloud_aws_instance$"
-            description = "Checks a single EC2 instance"
-        }
         "-i" = {
-            value = "$cloud_aws_id$"
+            value = "$cloud_aws_ec2_instance_id$"
```
[the remark this answers is not in the document]
Comment (pdorschner) [reply contrib/icinga2-commands.conf:31]: This is specific to the check command instance

File: contrib/icinga2-commands.conf [hunk 17]
```diff
@@ -8,81 +8,198 @@ template CheckCommand "cloud_aws" {
...
             description = "Look for ec2 instance by name"
         }
     }
-
-    vars.cloud_aws_ec2 = "ec2"
-    vars.loud_aws_instance = "instance"
 }
 
-object CheckCommand "cloud_aws_instance" {
+object CheckCommand "cloud_aws_instances" {
```
[the remark this answers is not in the document]
Comment (pdorschner) [reply contrib/icinga2-commands.conf:41]: Yes, it would be an duplicate

File: contrib/icinga2-commands.conf (hunk 4 above)
Comment (pdorschner) [reply contrib/icinga2-commands.conf:92]: Per default, if no buckets were found, it returns UNKNOWN. The use can decide if it returns OK or UNKNOWN

File: contrib/icinga2-commands.conf (hunk 9 above)
Comment (pdorschner) [reply contrib/icinga2-commands.conf:127]: Per default, if no buckets were found, it returns UNKNOWN. The use can decide if it returns OK or UNKNOWN

**Review by dgoetz — APPROVED** [review]
Looks good so far.
Personal preference would be having defaults as custom vars instead of or as addition to the description, but this is also valid.

**Pull request merged**